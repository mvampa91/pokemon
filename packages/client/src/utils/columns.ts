import { ReactNode } from "react"

interface ColumnsInput {
  nameRenderer?: RendererType,
  typesRenderer?: RendererType,
  classificationRenderer?: RendererType
}

type RendererType = (text: string[]) => ReactNode;

export const columns = ({
  nameRenderer,
  typesRenderer,
  classificationRenderer
}: ColumnsInput) => [
    {
      title: "Name",
      dataIndex: "name",
      key: "name",
      render: nameRenderer,
    },
    {
      title: "Type",
      dataIndex: "types",
      key: "types",
      render: typesRenderer,
    },
    {
      title: "Classification",
      dataIndex: "classification",
      key: "classification",
      render: classificationRenderer
    },
  ];