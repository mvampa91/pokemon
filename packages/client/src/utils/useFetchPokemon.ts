import { useQuery } from "@apollo/client";
import { DocumentNode } from "graphql";

export const useFetchPokemons = (query: DocumentNode, variables: { [key: string]: any }) => {
    const { loading, data } = useQuery(
        query,
        {
          variables,
        }
      );

      return {
          loading,
          data: data?.[variables.key]
      }
}
