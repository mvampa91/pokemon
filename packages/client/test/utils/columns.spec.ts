import { columns } from "../../src/utils/columns";


test('should return the desired structure', () => {
    const structure = columns({});

    expect(structure).toMatchSnapshot();
})

test('should return the structure with the desired rendered', () => {
    const myNameRenderer = jest.fn();
    const myTypesRenderer = jest.fn();
    const myClassificationRenderer = jest.fn();

    const structure = columns({
        nameRenderer: myNameRenderer,
        typesRenderer: myTypesRenderer,
        classificationRenderer: myClassificationRenderer,
    });

    expect(structure[0].render).toEqual(myNameRenderer);
    expect(structure[1].render).toEqual(myTypesRenderer);
    expect(structure[2].render).toEqual(myClassificationRenderer);
})

