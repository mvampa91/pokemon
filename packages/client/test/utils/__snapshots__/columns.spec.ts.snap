// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the desired structure 1`] = `
Array [
  Object {
    "dataIndex": "name",
    "key": "name",
    "render": undefined,
    "title": "Name",
  },
  Object {
    "dataIndex": "types",
    "key": "types",
    "render": undefined,
    "title": "Type",
  },
  Object {
    "dataIndex": "classification",
    "key": "classification",
    "render": undefined,
    "title": "Classification",
  },
]
`;
